packages/components/checkbox/src/Checkbox.tsx

```diff
@@ -9,6 +11,7 @@ import { CheckboxLabel } from './CheckboxLabel'
 export type CheckboxProps = CheckboxInputProps
 
 export const Checkbox = forwardRef<HTMLButtonElement, CheckboxProps>(
+  // eslint-disable-next-line complexity
```

Comment:
It's because of the `intent` const, replace with what we currently have and you won't need this eslint comment:
```tsx
const intent = (() => {
  if (isInvalid) return 'error'

  return intentProp ?? group.intent
})()
```

Or, you could make it a function:
```tsx
const getIntent = () => (isInvalid ? 'error' : intentProp ?? group.intent)
```

Reply:
Yes I noticed that :+1: But imho this syntax is less readable, and this is why I was looking for a better solution... But If nothing pops, I'll go back with your method :+1:

Reply:
@acd02 Just updated the code...wdyt? I guess it's mainly about personal preferences, but at least there is no more `eslint-disable-next-line complexity` :sweat_smile:

Reply:
Not really fond of the names, `getFieldOrGroupAttribute`, `getFieldOrGroupAttribute` 😄,
Why not go all the way and make something like:
```tsx
interface GetFieldAttributes {
  field: ReturnType<typeof useFormFieldState>
  group: ReturnType<typeof useCheckboxGroup>
  intent: CheckboxInputProps['intent']
}

function getFieldAttributes({ field, group, intent: intentProp }: GetFieldAttributes) {
  const name = field.name ?? group.name
  const isRequired = field.isRequired ?? group.isRequired
  const isInvalid = field.isInvalid ?? group.isInvalid
  const isFieldEnclosed = field.id !== group.id
  const id = isFieldEnclosed ? field.id : undefined
  const description = isFieldEnclosed ? field.description : undefined
  const intent = isInvalid ? 'error' : intentProp ?? group.intent

  return { name, isRequired, isInvalid, isFieldEnclosed, id, description, intent }
}

...

const { id, name, isInvalid, isRequired, description, intent } = getFieldAttributes({
  field,
  group,
  intent: intentProp,
})

...

return (
      <CheckboxLabel>
        <CheckboxInput
          ref={ref}
          id={id}
          name={name}
          value={value}
          intent={intent}
          checked={checked}
          disabled={disabled}
          required={isRequired}
          aria-describedby={description}
          aria-invalid={isInvalid}
          onCheckedChange={handleCheckedChange}
          {...others}
        />
        {children}
      </CheckboxLabel>
    )

```